colossalai/nn/layer/colossalai_layer/linear.py

```diff
@@ -65,15 +66,27 @@ def __init__(self,
             if self.layer.bias is not None:
                 bias_initializer(self.layer.bias, fan_in=in_features)
         else:
-            self.layer = _parallel_linear[tensor_parallel](
-                in_features,
-                out_features,
-                bias=bias,
-                dtype=dtype,
-                weight_initializer=weight_initializer,
-                bias_initializer=bias_initializer,
-                **kwargs,
-            )
+            if tensor_parallel == '1d': # gather_out arg is available

```

Comment:
We can actually use `inspect.signature` to check if gather_output is an argument of the specific linear module, for the purpose of avoiding writing the same codes twice.

Reply:
ok，I will change it.

Reply:
I use another way to eliminate double invoke. But I wonder how we can use inspect.signature to invoke function with different params?

Reply:
@kurisusnowdeng Done

Reply:
Done. @kurisusnowdeng